src/test/scala/query/MergeQuerySpec.scala

```diff
@@ -0,0 +1,159 @@
+package test.query
+
+import com.originate.scalypher.path.AnyNode
+import com.originate.scalypher.path.CypherNode
+import com.originate.scalypher.path.KindRelationship
+import com.originate.scalypher.MergeQuery
+
+import org.scalatest._
+
+class MergeQuerySpec extends WordSpec with Matchers {
+
+  val startNode = CypherNode("label1")
+  val createRelationship = KindRelationship("label")
+  val endNode = CypherNode("label2")
+  val matchPath = startNode -- endNode
+  val createPath = startNode -- createRelationship -- endNode
+  val where = startNode.property("id") <> "test"
+
+  "only a merge path is given" must {
+
+    "produce a query that merges a node" in {
+      val create = MergeQuery(startNode)
+      create.toQuery shouldBe "MERGE (:label1)"
+    }
+
+  }
+
+  "a match and merge path is given" must {
+
+    "use simple identifiers for references in the merge path" in {
+      val create = matchPath merge createPath
+      create.toQuery shouldBe "MATCH (a1:label1)--(a2:label2) MERGE (a1)-[:label]-(a2)"
+    }
+
+    "include a return statement if provided" in {
+      val create = matchPath merge createPath returnDistinct startNode
+      create.toQuery shouldBe """MATCH (a1:label1)--(a2:label2) MERGE (a1)-[:label]-(a2) RETURN DISTINCT a1"""
+    }
+
+    "add identifier to reference if it appears in the return statement" in {
+      val create = matchPath merge createPath returnDistinct createRelationship
+      val query = create.toQuery
+      query should include (")-[a1:label]-(")
+      query should endWith ("RETURN DISTINCT a1")
+    }
+
+    "allow multiple match paths and" must {
+
+      val otherNode = AnyNode()
+      val matchPath2 = startNode --> otherNode
+      val createPath = startNode -- createRelationship -- endNode --> otherNode
+
+      "assign identifiers to the necessary elements" in {
+        val merge = MergeQuery(createPath, Seq(matchPath, matchPath2))
+        val query = merge.toQuery
+        query should startWith regex """MATCH \(a\d:label1\)--\(a\d:label2\), \(a\d:label1\)-->\(a\d\)"""
+        query should endWith regex """MERGE \(a\d\)-\[:label]-\(a\d\)-->\(a\d\)"""
+      }
+
+      "assign identifiers appropriately" when {
+
+        val merge = MergeQuery(createPath, Seq(matchPath, matchPath2))
+          .onCreate(Seq(startNode.property("a").set("b")))
+          .onMerge(Seq(createRelationship.property("c").set("d"), otherNode.property("e").set("f")))
+
+        def extractIds(query: String) = {
+          val Identifiers = """MATCH \((\w+):label1\)--\((\w+):label2\), \((\w+):label1\)-->\((\w+)\).*""".r
+          val CreateIdentifier = """.*\[(\w*):label\].*""".r
+
+          query match {
```

Comment:
yeah i'm not a fan of the nested match, i think your suggestion cleans it up nicely. i can make startid1 == startid2 an assertion instead of a match error, that's a good point.